be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -683,37 +516,43 @@ Status JsonReader::_filter_row_with_jsonroot(simdjson::ondemand::object* row) {
 
 // read one json string from file read and parse it to json doc.
 Status JsonReader::_read_and_parse_json() {
+    uint8_t* data{};
+    size_t length = 0;
+
 #ifdef BE_TEST
+
     [[maybe_unused]] size_t message_size = 0;
     Slice result(_buf.data(), _buf_size);
     RETURN_IF_ERROR(_file->read(&result));
     if (result.size == 0) {
         return Status::EndOfFile("EOF of reading file");
     }
-    auto err = _parser.iterate_many(result.data, result.size, 32 * 1024 * 1024).get(_doc_stream);
+
+    data = reinterpret_cast<uint8_t*>(result.data);
+    length = result.size;
 
 #else
-    size_t length = 0;
+
     StreamPipeSequentialFile* stream_file = reinterpret_cast<StreamPipeSequentialFile*>(_file.get());
-    RETURN_IF_ERROR(stream_file->read_one_message(&_json_binary_ptr, &length));
+    // For efficiency reasons, simdjson requires a string with a few bytes (simdjson::SIMDJSON_PADDING) at the end.
+    RETURN_IF_ERROR(stream_file->read_one_message(&_json_binary_ptr, &length, simdjson::SIMDJSON_PADDING));
     if (length == 0) {
         return Status::EndOfFile("EOF of reading file");
     }
 
-    //TODO: Pre-allocate 200MB memory. We could process big array in stream.
-    auto err = _parser.iterate_many(_json_binary_ptr.get(), length, 200 * 1024 * 1024).get(_doc_stream);
+    data = _json_binary_ptr.get();
+
 #endif
 
-    if (err) {
-        std::string err_msg = strings::Substitute("Failed to parse string to json. code=$0, error=$1", err,
-                                                  simdjson::error_message(err));
-        return Status::DataQualityError(err_msg.c_str());
+    // If _strip_outer_array == true, we try to iterate the json as array.
+    if (_scanner->_strip_outer_array) {
+        _parser.reset(new JsonArrayParser);
```

Comment:
What's the difference between JsonArrayParser and JsonDocumentStreamParser?

Reply:
`JsonArrayParser` using `simdjson::ondemand::parser::iterate` to parse json array, which only supports parsing single document in streaming.
`JsonDocumentStreamParser` using `simdjson::ondemand::parser::iterate_many` to parse ndjson.

Reply:
OK. Do you use strip_outer_array to check whether the input is a ndjson or not?

Reply:
The `strip_outer_array` is checked before the `JsonParser` is constructed.Then, the json input format is checked in `JsonParser::parser()`.